ttnn/cpp/ttnn/decorators.hpp

```diff
@@ -162,48 +133,41 @@ struct operation_t {
 
-            auto output_tensors = create_async_output_tensors<concrete_operation_t>(
+            auto output_tensors = detail::create_async_output_tensors<concrete_operation_t>(
                 std::move(input_tensors), std::move(optional_input_tensors), std::forward<Args>(args)...);
 
-            auto remaining_args =
-                extract_args_excluding_types_by_value<ttnn::Tensor, std::optional<const ttnn::Tensor>>(
-                    std::forward<Args>(args)...);
-
             // TODO: add support for optional_output_tensors
             // auto optional_output_tensors = extract_args_to_vector(std::forward<Args>(args)...,
             // std::optional<ttnn::Tensor>);
 
             bool enable_autoformat = false;
             operation::launch_op(
-                [cpp_fully_qualified_name = this->cpp_fully_qualified_name, remaining_args](
+                [cpp_fully_qualified_name = this->cpp_fully_qualified_name, args...](
                     const Tensors& input_tensors,
                     const OptionalConstTensors& optional_input_tensors,
                     const OptionalTensors&) mutable -> Tensors {
-                    tt::log_debug(tt::LogOp, "Launching C++ ttnn operation in async: {}", cpp_fully_qualified_name);
-
+                    tt::log_debug(
+                        tt::LogOp, "Launching C++ ttnn operation in async mode: {}", cpp_fully_qualified_name);
+                    auto execute_args = detail::map_launch_op_args_to_execute(
+                        input_tensors, optional_input_tensors, std::forward<Args>(args)...);
                     return std::apply(
-                        [cpp_fully_qualified_name, &input_tensors, &optional_input_tensors](auto&&... args) -> Tensors {
-                            auto args_execute_tuple = concrete_operation_t::map_launch_op_args_to_execute(
-                                input_tensors, optional_input_tensors, std::forward<decltype(args)>(args)...);
-                            return std::apply(
-                                [cpp_fully_qualified_name](auto&&... args) -> Tensors {
-                                    detail::validate<concrete_operation_t>(
-                                        cpp_fully_qualified_name, std::forward<decltype(args)>(args)...);
-                                    return detail::map_execute_return_to_vector<concrete_operation_t>(
-                                        concrete_operation_t::execute(std::forward<decltype(args)>(args)...));
-                                },
-                                args_execute_tuple);
+                        [cpp_fully_qualified_name](auto&&... args) -> Tensors {
+                            detail::validate<concrete_operation_t>(
+                                cpp_fully_qualified_name, std::forward<decltype(args)>(args)...);
+                            return detail::map_execute_return_to_vector<concrete_operation_t>(
```

Comment:
When you are inside launc_op callback you are on a different thread reaching back to the args on the operator() overload.  This means that the remaining args where not passed by value and so you end up with a bug here.